src/cc/arduino/plugins/wifi101/flashers/java/NinaFlasher.java

```diff
@@ -0,0 +1,338 @@
+import java.util.Arrays;
+import java.util.*;
+import java.util.List;
+import java.io.ByteArrayOutputStream;
+import java.io.*;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import org.apache.commons.lang3.StringUtils;
+import java.security.MessageDigest;
+
+import cc.arduino.plugins.wifi101.certs.WiFi101Certificate;
+import cc.arduino.plugins.wifi101.certs.WiFi101CertificateBundle;
+
+import cc.arduino.plugins.wifi101.flashers.Flasher;
+import javax.swing.JProgressBar;
+
+public class NinaFlasher implements Flasher {
+  public String boardName;
+  public String version;
+  public File file;
+  public File Compatiblefile;
+  public JProgressBar progressBar;
+  public String name;
+  public List<String> compatibleboard;
+  public byte[] md5Checksum;
+
+  public NinaFlasher(String _name, String _version, String _filename) {
+    boardName = _name;
+    compatibleboard =  new ArrayList<String>();
+    version = _version;
+    file = null;
+    name = "WifiNINA";
+
+    try {
+      String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+      File jarFolder = new File(jarPath).getParentFile();
+      file = new File(jarFolder, _filename);
+      md5Checksum = getMD5Checksum(file.getPath());
+    }
+    catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+    catch (Exception e) {
+      e.printStackTrace();
+    }
+    try {
+      String line = null;
+      String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+      File jarFolder = new File(jarPath).getParentFile();
+      Compatiblefile = new File(jarFolder, "firmwares/Config.ini");
+      BufferedReader bufferedReader = new BufferedReader(new FileReader(Compatiblefile));
+      while ((line = bufferedReader.readLine()) != null) {
+        if (name.equals(line.substring(line.indexOf("Name") + 4, line.indexOf("BF")))) {
+          if (boardName.equals(line.substring(line.indexOf("BF")+2,line.indexOf("Version")))) {
+            if (version.equals(line.substring(line.indexOf("Version") + 7, line.indexOf("Boards")))) {
+              line = line.substring(line.indexOf("Boards") + 6, line.indexOf(":") + 1);
+              int old = 0;
+              while (line.substring(old, line.indexOf(":")) != null) {
+                if (line.indexOf(";") < 0) {
+                  compatibleboard.add(line.substring(old, line.indexOf(":")));
+                  break;
+                } else {
+                  compatibleboard.add(line.substring(old, line.indexOf(";")));
+                  old = line.indexOf(";") + 1;
+                  line = line.substring(old, line.indexOf(":") + 1);
+                  old = 0;
+                }
+              }
+            }
+          }
+        }
+      }
+      bufferedReader.close();
+    }
+    catch (URISyntaxException e) {
+      e.printStackTrace();
```

Comment:
is this how we want to handle errors? It's also inconsistent will the other exceptions below ...